week2/project/ecommerce2/src/components/ProductDetails.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import ErrorMsg from './Error';
+
+export default function ProductDetails() {
+  const { id } = useParams();
+  const [product, setProduct] = useState([]);
```

Comment:
why making inial state is empty array ? then you well set it as object when you get single product.

the proper way is to make it null or let it be undefined.
✏️need to modify✏️

Reply:
fixed. made useState(0)